samples/event-sourcing-esdb-simple/src/main/java/io/eventdriven/ecommerce/shoppingcarts/productitems/ProductItem.java

```diff
@@ -0,0 +1,18 @@
+package io.eventdriven.ecommerce.shoppingcarts.productitems;
+
+import java.util.UUID;
+
+public record ProductItem(
+  UUID productId,
+  int quantity
+) {
+  public static ProductItem of(UUID productId, Integer quantity) {
+    if (productId == null)
+      throw new IllegalArgumentException("ProductId has to be defined");
+
+    if (quantity == null || quantity <= 0)
```

Comment:
Please create a dedicated self validating object called Quantity - Value Object knows from [DDD](https://medium.com/swlh/value-objects-to-the-rescue-28c563ad97c6)

Reply:
I think that for this small case it should be enough to wrap it in the ProductItem validation, as I won't be using it in other places (for now).